parcels/gridset.py

```diff
@@ -40,10 +40,14 @@ def dimrange(self, dim):
            in a gridset. Useful for finding e.g. longitude range that
            overlaps on all grids in a gridset"""
 
-        maxleft, minright = (0, np.infty)
+        maxleft, minright = (0, np.inf)
         for g in self.grids:
-            maxleft = max(maxleft, getattr(g, dim)[0])
-            minright = min(minright, getattr(g, dim)[-1])
+            if dim == 'time_full' and len(getattr(g, dim)) == 1:
+                continue  # not including grids where only one time_full entry, as with allow_time_extrapolation=True
```

Comment:
Should we do exactly the same for depth dimension?

This can be interesting if we want to read both 2d and 3d fields
(example velocity 3d and EEZ 2d)